shared_models/shared_models.py

```diff
@@ -111,15 +111,18 @@ class PresentationExchange(BaseModel):
     presentation_request: Optional[IndyProofRequest] = None
     protocol_version: ProofRequestProtocolVersion
     role: Literal["prover", "verifier"]
-    state: Literal[
-        "proposal-sent",
-        "proposal-received",
-        "request-sent",
-        "request-received",
-        "presentation-sent",
-        "presentation-received",
-        "done",
-        "abandoned",
+    state: Union[
+        None,
+        Literal[
+            "proposal-sent",
+            "proposal-received",
+            "request-sent",
+            "request-received",
+            "presentation-sent",
+            "presentation-received",
+            "done",
+            "abandoned",
+        ],
```

Comment:
State is never None I think?

Reply:
Added this recently. Much to my surprise it can be None/was None. I think that can happen when errors occur in aca-py. I'm not entirely sure how/why this occurs/occured but I'm sure it did occur. The annoying thing when leaving this out was that also the webhooks OpenApi stopped working when it occured because obviously the models check throws a validation error.